### PR TITLE
Master

### PR DESCRIPTION
**Pull request by Brummell67**

### CODE REVIEW

**Comment by Brummell67** [comment]
In this commit, there is also modify for open "CairoSettingsWindow.xaml" as Dialog.

**Comment by josuave** [comment]
Should this setting show if the has has cairo set as shell. It seems that this should only be available if cairo is not shell. Other considerations, cairo has been launched by another startup location?????

**Comment by Brummell67** [comment]
> Should this setting show if the has has cairo set as shell. It seems that this should only be available if cairo is not shell. Other considerations, cairo has been launched by another startup location?????

Can you explain better?
I'm noob, exceuse me.

File: Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml [hunk 1]
```diff
@@ -172,6 +177,10 @@
                         Name="chkEnableAutoUpdates">
                         <Label Content="{Binding Path=(l10n:DisplayString.sSettings_General_UpdateCheck)}" />
                     </CheckBox>
+                    <CheckBox Click="ChkRunAtLogOn_Click"
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml]: Private methods in Cairo use a lower case first letter. Please follow this convention and revert the capitalizations

File: Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs [hunk 2]
```diff
@@ -22,18 +22,18 @@ public CairoSettingsWindow()
         {
             InitializeComponent();
 
-            loadThemes();
-            loadLanguages();
-            loadRadioGroups();
-            loadCategories();
-            loadHotKeys();
-            loadLoggingLevels();
+            LoadThemes();
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs]: As above, private methods in Cairo use a lowercase first letter. Please revert the capitalizations

File: Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs [hunk 3]
```diff
@@ -392,5 +392,24 @@ private void btnChangeShell_Click(object sender, RoutedEventArgs e)
...
+            if (chkBox.IsChecked.Equals(false))
+            {
+                //Delete SubKey
+                key.DeleteValue("Cairo");
+            }
+            else
+            {
+                //Write SubKey
+                //Settings.RunAtLogOn = true;
+                key.SetValue("Cairo", @"C:\Program Files\Cairo Shell\CairoDesktop.exe");
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs]: This makes an assumption that Cairo is installed in this location. It should use the path of the current running executable.

File: Cairo Desktop/Cairo Desktop/MenuBar.xaml.cs [hunk 4]
```diff
@@ -863,7 +863,7 @@ private void SysSleep(object sender, RoutedEventArgs e)
         private void InitCairoSettingsWindow(object sender, RoutedEventArgs e)
         {
             CairoSettingsWindow window = new CairoSettingsWindow();
-            window.Show();
+            window.ShowDialog();
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/MenuBar.xaml.cs]: ShowDialog() is a blocking operation and should be using Show() as it was previously.

File: Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml [hunk 5]
```diff
@@ -172,6 +177,10 @@
...
+                        x:Name="chkRunAtLogOn" IsChecked="{Binding RunAtLogOn}">
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml]: It is reading its checked state from a stored setting, but this is not sustainable, since it can be mis-matched if another program modifies the registry. It should check the registry and not be a stored setting.

File: Cairo Desktop/CairoDesktop.Configuration/Properties/Settings.settings [hunk 6]
```diff
@@ -98,5 +98,8 @@
     <Setting Name="FullWidthTaskBar" Type="System.Boolean" Scope="User">
       <Value Profile="(Default)">False</Value>
     </Setting>
+    <Setting Name="RunAtLogOn" Type="System.Boolean" Scope="User">
```
Comment (dremin) [inline Cairo Desktop/CairoDesktop.Configuration/Properties/Settings.settings]: This should not be a saved setting, remove this and instead read from the system registry.

File: Cairo Desktop/CairoDesktop.Configuration/CairoDesktop.Configuration.csproj [hunk 7]
```diff
@@ -67,6 +67,7 @@
     </Compile>
     <Compile Include="Settings.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="Settings1.cs" />
```
Comment (dremin) [inline Cairo Desktop/CairoDesktop.Configuration/CairoDesktop.Configuration.csproj]: Why was this file added?

File: Cairo Desktop/CairoDesktop.Configuration/Settings.cs [hunk 8]
```diff
@@ -108,6 +109,23 @@ public static bool IsFirstRun
             }
         }
 
+        public static bool RunAtLogOn
```
Comment (dremin) [inline Cairo Desktop/CairoDesktop.Configuration/Settings.cs]: Since this shouldn't be a stored setting, references should be removed from this file as well.

Comment (dremin) [inline Cairo Desktop/CairoDesktop.Configuration/Settings.cs]: Since this shouldn't be a stored setting, references should be removed from this file as well.

File: Cairo Desktop/CairoDesktop.Configuration/Settings1.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
+﻿namespace CairoDesktop.Configuration.Properties {
```
Comment (dremin) [inline Cairo Desktop/CairoDesktop.Configuration/Settings1.cs]: I think this file can be removed as per above.

File: Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml [hunk 10]
```diff
@@ -172,6 +177,10 @@
...
+                        <Label Content="{Binding Path=(l10n:DisplayString.sRunAtLogon)}" />
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml]: The setting should only be visible if Cairo is not currently the running shell. You can view the other code that checks this for reference.

File: Cairo Desktop/CairoDesktop.Localization/Language.cs [hunk 11]
```diff
@@ -6,6 +6,7 @@ static class Language
     {
         public static Dictionary<string, string> en_US = new Dictionary<string, string>
         {
+            { "sRunAtLogon", "Run at log-on" },
```
Comment (dremin) [inline Cairo Desktop/CairoDesktop.Localization/Language.cs]: Is this a duplicate of sSettings_General_RunAtLogOn?

File: Cairo Desktop/CairoDesktop.Localization/DisplayString.cs [hunk 12]
```diff
@@ -84,6 +84,14 @@ private static string getString(string stringName)
             return stringName;
         }
 
+        public static string sRunAtLogon
+        {
+            get
+            {
+                return getString("sRunAtLogon");
```
Comment (dremin) [inline Cairo Desktop/CairoDesktop.Localization/DisplayString.cs]: Is this a duplicate of sSettings_General_RunAtLogOn?

File: Cairo Desktop/CairoDesktop.Configuration/app.config [hunk 13]
```diff
@@ -103,6 +103,9 @@
             <setting name="FullWidthTaskBar" serializeAs="String">
                 <value>False</value>
             </setting>
+            <setting name="RunAtLogOn" serializeAs="String">
```
Comment (dremin) [inline Cairo Desktop/CairoDesktop.Configuration/app.config]: Since this is no longer being stored in the settings but rather checked via the registry, can this `<setting>` be removed from the project files?

File: Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs [hunk 14]
```diff
@@ -392,5 +394,44 @@ private void btnChangeShell_Click(object sender, RoutedEventArgs e)
...
+            if (chkBox.IsChecked.Equals(false))
+            {
+                //Delete SubKey
+                rKey.DeleteValue("Cairo");
+            }
+            else
+            {
+                string exePath = System.Reflection.Assembly.GetExecutingAssembly().Location;
+                //Write SubKey
+                rKey.SetValue("Cairo", exePath);
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs]: The installer sets the name as "CairoShell". This should be made consistent with that.

File: Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs [hunk 15]
```diff
@@ -392,5 +394,44 @@ private void btnChangeShell_Click(object sender, RoutedEventArgs e)
...
+        {
+            if (Shell.IsCairoUserShell.Equals(true))
+            {
+                chkRunAtLogOn.Visibility = Visibility.Hidden;
+            }
+
+            RegistryKey rKey = Registry.CurrentUser.CreateSubKey(@"Software\Microsoft\Windows\CurrentVersion\Run");
+            List<string> rKeyValueNames = rKey.GetValueNames().ToList();
+
+            if (rKeyValueNames.Contains("Cairo"))
```
Comment (dremin) [inline Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs]: Since the installer sets this to CairoShell, I don't think this check will work unless it is changed to look for "CairoShell".
